### PR TITLE
Make projectiles not hit crates unless clicked on

### PR DESCRIPTION
**Pull request by Cojoke-dot**
## About the PR

Title
## Why / Balance

Consistency for shorter objects while not providing a spot someone can just become invincible to gunfire.
## Technical details

Creates a new system and component called RequireTarget. Works on the systems implemented in #27905 
## Media

https://github.com/space-wizards/space-station-14/assets/83733158/61eaeb5e-b184-4e05-8533-04b6debd88b9

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl:
- tweak: Bullets no longer hit crates unless directly clicked on while shooting.
- tweak: Bullets now pass over mobs that are lying down unless directly clicked on while shooting.

### CODE REVIEW

**Review by AJCM-git — APPROVED** [review]
.

**Comment by Cojoke-dot** [comment]
I'm not sure If I did it in the way you meant but it seems to work. Using StandingState means that bullets pass over the heads of people lying down, which makes sense I guess( I'll add that to the change log if you would like). There are three issues I see with having it like that:
1. slipping someone then mag dumping them is harder
2. laying on a bed to dodge bullets
3. cutting off your legs with newmed so you lay on the floor( depends on newmed of course)

I like the idea of laying and bullets passing overhead, so I did not add something to check if they were crit, but I would understand if that's not something that's desired for the game.

File: Content.Shared/Damage/Systems/RequireProjectileTargetSystem.cs
Comment (AJCM-git) [inline Content.Shared/Damage/Systems/RequireProjectileTargetSystem.cs:1]: use Entity<> in the event handlers

File: Content.Shared/Damage/Components/RequireProjectileTargetComponent.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+namespace Content.Shared.Damage.Components;
+
+/// <summary>
+/// Prevent the object from getting hit by projetiles unless you target the object.
+/// </summary>
+[RegisterComponent]
```
Comment (AJCM-git) [inline Content.Shared/Damage/Components/RequireProjectileTargetComponent.cs:6]: networked component, access and also network Active with AutoNetworkedField attribute and dirty when you update it (add a `SetActive(Entity<COMPONENT> ent, bool value) { if (ent.Comp.Active == value) return; ent.Comp.Active = value; Dirty(ent)}` method)

**Review by AJCM-git — CHANGES_REQUESTED** [review]
> 1. slipping someone then mag dumping them is harder

Thats probably fine

> 2. laying on a bed to dodge bullets

thats almost certain death, worst case if it somehow gets abused we can just add a delay to getting in/out of bed

> 3. cutting off your legs with newmed so you lay on the floor( depends on newmed of course)

Certain death

If it ends up a problem in the future we can add a datafield that checks for the knockeddown status effect in prevent collide so we can swap at will tho i doubt that will be needed

Also, another easy thing to do here or next PR would be adding this component to `BaseComputer` (and pretty much every machine that doesnt "look" big)

Remember to update CL

**Review by AJCM-git — APPROVED** [review]

**Pull request merged**